src/sclient/StorageClient.cpp

```diff
@@ -0,0 +1,235 @@
+
+    auto spaceVidLenResult = mclient_->getSpaceVidLen(spaceId);
+    if (!spaceVidLenResult.ok()) {
+        LOG(ERROR) << "Get space vid len for " << spaceName
+                   << " failed: " << spaceVidLenResult.status();
+        return -1;
+    }
+    return spaceVidLenResult.value();
+}
+
+std::vector<int32_t> StorageClient::getParts(std::string spaceName) {
+    std::vector<int32_t> parts;
+
+    auto partsAlloc = this->getPartsAlloc(spaceName);
+
+    parts.reserve(partsAlloc.size());
+
+    for (auto &partAlloc : partsAlloc) {
+        parts.push_back(partAlloc.first);
+    }
+    return parts;
+}
+
+std::unordered_map<int32_t, std::vector<nebula::HostAddr>> StorageClient::getPartsAlloc(
+    std::string spaceName) {
+    std::unordered_map<int32_t, std::vector<nebula::HostAddr>> result;
+
+    auto spaceIdResult = mclient_->getSpaceIdByNameFromCache(spaceName);
+    if (!spaceIdResult.ok()) {
+        LOG(ERROR) << "Get space id for " << spaceName << " failed: " << spaceIdResult.status();
+        return result;
+    }
+    auto spaceId = spaceIdResult.value();
+
+    auto partsAllocResult = mclient_->getPartsAlloc(spaceId).get();
+    if (!partsAllocResult.ok()) {
+        LOG(ERROR) << "Get parts alloc for space: " << spaceName
+                   << " failed: " << partsAllocResult.status();
+        return result;
+    }
+    result = partsAllocResult.value();
+    return result;
+}
+
+std::unordered_map<int32_t, nebula::HostAddr> StorageClient::getPartsLeader(std::string spaceName) {
+    std::unordered_map<int32_t, nebula::HostAddr> result;
+
+    auto spaceIdResult = mclient_->getSpaceIdByNameFromCache(spaceName);
+    if (!spaceIdResult.ok()) {
+        LOG(ERROR) << "Get space id for " << spaceName << " failed: " << spaceIdResult.status();
+        return result;
+    }
+    auto spaceId = spaceIdResult.value();
+
+    auto partsAllocResult = mclient_->getPartsAlloc(spaceId).get();
+    if (!partsAllocResult.ok()) {
+        LOG(ERROR) << "Get parts alloc for space: " << spaceName
+                   << " failed: " << partsAllocResult.status();
+        return result;
+    }
+    auto partsAlloc = partsAllocResult.value();
+    for (auto &partAlloc : partsAlloc) {
+        auto partId = partAlloc.first;
+        auto leaderResult = sclient_->getLeader(spaceId, partId);
+        if (!leaderResult.ok()) {
+            LOG(ERROR) << "Get leader host for space: " << spaceName << ", part: " << partId
+                       << " failed: " << leaderResult.status();
+            return result;
+        }
+        result[partId] = leaderResult.value();
+    }
+    DCHECK(result.size() == partsAlloc.size());
+
+    return result;
+}
+
+ScanEdgeIter StorageClient::scanEdgeWithPart(std::string spaceName,
+                                             int32_t partId,
+                                             std::string edgeName,
+                                             std::vector<std::string> propNames,
```

Comment:
the keys of the edge is unknown by users, you need to add the key prop of the edge.

```
'_src'
'_type'
'_rank'
'_dst'
```

Reply:
currently I just need the "_src" and "_dst"